### ISSUE TITLE
meta.maintainers for modules

### ISSUE THREAD

**Issue by roberth** [LnL7/nix-darwin#291]
The module system supports a `meta` attribute in the module syntax that adds to the `meta.<...>` options, but I don't see that feature used in nix-darwin. I could open a PR based on NixOS's solution if you think it makes sense to use this feature.

**Comment by LnL7** [comment]
Sounds good to me. Discrepancies with the nixos modules are generally either things I omitted for simplicity since it wasn't super relevant or things that changed since the initial implementation.

The only thing to take in consideration is that unlike on nixos using the maintainer list from nixpkgs here can't be guaranteed to be consistent.

**Comment by roberth** [comment]
Good point. Perhaps they should be entered as github handle strings instead? That way we can fall back to partial data instead of throwing an exception. We can still use the Nixpkgs maintainer database to enrich the github handles in `mkOption`'s `apply`.

**Comment by LnL7** [comment]
As long as regular system activation doesn't have to evaluate this it's not a big deal, I don't know where this is actually evaluated on nixos.

**Comment by roberth** [comment]
For packages it'd evaluated on hydra. I don't know about any programmed use of `meta.maintainers` either.
Most importantly it can be read by humans in the source.

The accumulated data looks like this

```
nix-repl> :p (nixos { fileSystems."/".device = "x"; boot.loader.grub.enable = false; } ).config.meta.maintainers
{ "/home/user/nixpkgs/nixos/modules/config/malloc.nix" = [ { email = "joachifm@fastmail.fm"; github = "joachifm"; githubId = 41977; name = "Joachim Fasting"; } ]; "/home/user/nixpkgs/nixos/modules/config/vte.nix" = [ #...
```

**Comment by LnL7** [comment]
If it's lazy then possible inconsistencies aren't a big deal. The main case to avoid is that people can't rebuild their system after updating nix-darwin without also having to pull in a newer version of nixpkgs.

**Comment by Enzime** [comment]
Implemented in #586
